lib/events/dynamoevents/dynamoevents.go

```diff
@@ -890,81 +920,96 @@ func (l *Log) indexExists(tableName, indexName string) (bool, error) {
 	return false, nil
 }
 
-// createV2GSI creates the new global secondary Index and updates
+// createV2GSI creates the new global secondary index if it does not exist and updates
 // the schema to add a string key `date`.
 //
 // This does not remove the old global secondary index.
 // This must be done at a later point in time when all events have been migrated as per RFD 24.
 //
 // Invariants:
-// - The new global secondary index must not exist.
-// - This function must not be called concurrently with itself.
 // - This function must be called before the
 //   backend is considered initialized and the main Teleport process is started.
 func (l *Log) createV2GSI(ctx context.Context) error {
-	provisionedThroughput := dynamodb.ProvisionedThroughput{
-		ReadCapacityUnits:  aws.Int64(l.ReadCapacityUnits),
-		WriteCapacityUnits: aws.Int64(l.WriteCapacityUnits),
-	}
+	err := backend.RunWhileLocked(ctx, l.backend, indexV2CreationLock, 5*time.Minute, func(ctx context.Context) error {
```

Comment:
Nit: Right now we are having:
```
func createV2GSI{
        err := backend.RunWhileLocked{
             //  createV2GSI function logic. 
         }
    return trace.Wrap(err)
 }
```
What about moving  the createV2GSI  call inside a lock: 
  ```
          err := backend.RunWhileLocked(..., func() error {
              return trace.Wrap(l.createV2GSI(....))
         }
```

Reply:
Done